sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,934 @@
+
+    return executeAtomXmlOperation(this, webResource, serializer);
+  }
+
+  /**
+   * Deletes a resource.
+   * @param name
+   */
+  private async deleteResource(
+    name: string,
+    serializer: AtomXmlSerializer
+  ): Promise<HttpOperationResponse> {
+    const webResource: WebResource = new WebResource(this.getUrl(name), "DELETE");
+
+    return executeAtomXmlOperation(this, webResource, serializer);
+  }
+
+  private getUrl(path: string, queryParams?: { [key: string]: string }): string {
+    const baseUri = `https://${this.endpoint}/${path}`;
+
+    const requestUrl: URLBuilder = URLBuilder.parse(baseUri);
+    requestUrl.setQueryParameter(Constants.API_VERSION_QUERY_KEY, Constants.CURRENT_API_VERSION);
+
+    if (queryParams) {
+      for (const key of Object.keys(queryParams)) {
+        requestUrl.setQueryParameter(key, queryParams[key]);
+      }
+    }
+
+    return requestUrl.toString();
+  }
+
+  private static parseConnectionString(connectionString: string): { [key: string]: string } {
+    const output: { [key: string]: string } = {};
+    const parts = connectionString.trim().split(";");
+
+    for (let part of parts) {
+      part = part.trim();
+
+      if (part === "") {
+        // parts can be empty
+        continue;
+      }
+
+      const splitIndex = part.indexOf("=");
+      if (splitIndex === -1) {
+        throw new Error(
+          "Connection string malformed: each part of the connection string must have an `=` assignment."
+        );
+      }
+
+      const key = part.substring(0, splitIndex).trim();
+      if (key === "") {
+        throw new Error("Connection string malformed: missing key for assignment");
+      }
+
+      const value = part.substring(splitIndex + 1).trim();
+
+      output[key] = value;
+    }
+
+    return output;
+  }
+
+  private getSubscriptionPath(topicName: string, subscriptionName: string): string {
+    return topicName + "/Subscriptions/" + subscriptionName;
+  }
+
+  private getRulePath(topicName: string, subscriptionName: string, ruleName: string): string {
+    return topicName + "/Subscriptions/" + subscriptionName + "/Rules/" + ruleName;
+  }
+
+  getDeadLetterPath(queueName: string): string {
+    return `${queueName}/$DeadLetterQueue`;
+  }
+
+  private buildListQueuesResponse(response: HttpOperationResponse): ListQueuesResponse {
+    const queues: Queue[] = [];
+    const rawQueueArray: any = response.parsedBody || [];
+    for (let i = 0; i < rawQueueArray.length; i++) {
+      const queue = buildQueue(rawQueueArray[i]);
+      if (queue) {
+        queues.push(queue);
+      }
+    }
+    const listQueuesResponse: ListQueuesResponse = Object.assign(queues, { _response: response });
+    return listQueuesResponse;
+  }
+
+  private buildQueueResponse(response: HttpOperationResponse): QueueResponse {
+    const queue = buildQueue(response.parsedBody);
+    const queueResponse: QueueResponse = Object.assign(queue || {}, { _response: response });
```

Comment:
Yes response.parsedBody can be undefined and this is the default type right?
Functionality/scenarios wise, empty get/list requests can end up with undefined parsed body iirc.
It doesn't seem like parsedBody being undefined should necessarily be an error scenario

Reply:
Yes, `response.parsedBody` is allowed to be undefined. My question was more in the context of request/responses that we are dealing with atom xml apis for service bus. Do we expect all successful requests to get a response with a non empty body? If not, then are we aware of which cases can have empty response body?

The reason I ask is that empty body does indicate an error scenario for certain requests like get, create or update where the user would expect the response to have queue related information. In such cases, if we do get empty response, it can be undesirable to treat it as a success.

Reply:
For list requests, the response can be empty.
For get too technically they can be valid empty responses if they don't exist.

Reply:
`buildQueueResponse()` does not get used for list requests.
It is used for create, get, update and delete requests.
In all these scenarios, our current expectation is to receive a non empty body, correct?

If yes, then masking a potential issue where an empty body is received doesn't seem like the right thing to do.

Please check with @bterlson on what is the best approach in such cases.

Reply:
Resolving this comment thread for now.
This is ok if our response to the user has only optional fields.
Once we have mandatory fields, this approach will not scale.
Logged #5925 for further disucssion